### PR TITLE
Update PSReadLine corresponding to the prediction interface update

### LINKED ISSUE #2221: PSReadLine doesn't work on 7.2.0-preview.3
Ref: PowerShell/PowerShell#14649
Cc: @daxian-dbw

## Environment

```none
PS version: 7.2.0-preview.3
PSReadline version: 2.2.0-beta1
os: 10.0.21313.1000 (WinBuild.160101.0800)
PS file version: 7.2.0.0
HostName: ConsoleHost (Windows Terminal)
BufferWidth: 314
BufferHeight: 62
```

## Exception report

```
System.MissingMethodException: Method not found: 'Void System.Management.Automation.Subsystem.CommandPrediction.OnCommandLineAccepted(System.Collections.Generic.IReadOnlyList`1<System.String>)'.
   at Microsoft.PowerShell.PSConsoleReadLine.Microsoft.PowerShell.Internal.IPSConsoleReadLineMockableMethods.OnCommandLineAccepted(IReadOnlyList`1 history)
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics, CancellationToken cancellationToken)
```

## Steps to reproduce

Try running anything. Cannot even `cd` into directory, exception on `Enter` key press, aborting execution.

## Expected behavior

Working PowerShell, no exceptions.

## Actual behavior

Exception, not working command prediction and even basic stuff.

### PR DESCRIPTION
**Pull request by daxian-dbw**
# PR Summary

Fix #2221

Major changes:
1. Update PSReadLine corresponding to the prediction interface update
    - provide the `OnSuggestionDisplayed` feedback
    - pass in the client-id `PSReadLine`, as well as the predictor mini-session id, when providing feedback.
2. Move to net6.0 and consume `PowerShell 7.2.0-preview.3` NuGet packages

## PR Checklist

- [x] PR has a meaningful title
    - Use the present tense and imperative mood when describing your changes
- [x] Summarized changes
- [x] Make sure you've added one or more new tests
- [x] Make sure you've tested these changes in terminals that PowerShell is commonly used in (i.e. conhost.exe, Windows Terminal, Visual Studio Code Integrated Terminal, etc.)
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] Documentation needed at [PowerShell-Docs](https://github.com/MicrosoftDocs/PowerShell-Docs)
        - [ ] Doc Issue filed: 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/PowerShell/PSReadLine/pull/2225)

### CODE REVIEW

File: PSReadLine/Prediction.Entry.cs [hunk 1]
```diff
@@ -15,14 +15,21 @@ public partial class PSConsoleReadLine
...
         {
             internal readonly Guid PredictorId;
+            internal readonly uint? PredictorSession;
             internal readonly string Source;
             internal readonly string SuggestionText;
             internal readonly int InputMatchIndex;
 
-            internal SuggestionEntry(string soruce, Guid id, string suggestion, int matchIndex)
+            internal SuggestionEntry(string suggestion, int matchIndex)
+                : this(soruce: "History", predictorId: Guid.Empty, predictorSession: null, suggestion, matchIndex)
```
Comment (as-kien) [inline PSReadLine/Prediction.Entry.cs]: should it be `source`?

Comment (daxian-dbw) [reply PSReadLine/Prediction.Entry.cs]: Thanks for catching the typo!

File: test/InlinePredictionTest.cs [hunk 2]
```diff
@@ -395,9 +395,9 @@ internal static List<PredictionResult> MockedPredictInput(Ast ast, Token[] token
             return new List<PredictionResult>
             {
                 (PredictionResult)ctor.Invoke(
-                    new object[] { predictorId_1, "TestPredictor", suggestions_1 }),
+                    new object[] { predictorId_1, "TestPredictor", (uint)56, suggestions_1 }),
```
Comment (SteveL-MSFT) [inline test/InlinePredictionTest.cs]: should `56` be a const defined somewhere?  It's used in multiple places

File: MockPSConsole/MockPSConsole.csproj [hunk 3]
```diff
@@ -17,8 +17,8 @@
     <PackageReference Include="PowerShellStandard.Library" version="5.1.0" />
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFramework)' == 'net5.0'">
-    <PackageReference Include="Microsoft.PowerShell.SDK" version="7.1.0" />
+  <ItemGroup Condition="'$(TargetFramework)' == 'net6.0'">
+    <PackageReference Include="Microsoft.PowerShell.SDK" version="7.2.0-preview.3" />
```
Comment (SteveL-MSFT) [inline MockPSConsole/MockPSConsole.csproj:21]: What error does the user get if they try to use plugin with PS < 7.2?

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: MockPSConsole/MockPSConsole.csproj (hunk 3 above)
Comment (daxian-dbw) [reply MockPSConsole/MockPSConsole.csproj:21]: They will get the following error:
```
PS:4> Set-PSReadLineOption -PredictionSource HistoryAndPlugin
Set-PSReadLineOption: The prediction plugin source is not supported in this version of PowerShell. The 7.2 or a higher version of PowerShell is required to use this source.
```

File: test/InlinePredictionTest.cs (hunk 2 above)
Comment (daxian-dbw) [reply test/InlinePredictionTest.cs]: Yep, will do.

**Review by adityapatwardhan — APPROVED** [review]

File: test/InlinePredictionTest.cs (hunk 2 above)
Comment (daxian-dbw) [reply test/InlinePredictionTest.cs]: Feedback was addressed. Please take another look when you have time, thanks!

File: test/ListPredictionTest.cs [hunk 4]
```diff
@@ -1052,6 +1069,9 @@ public void List_PluginSource_Acceptance()
                         NextLine,
                         TokenClassification.None, new string(' ', windowWidth)
                      )),
+                // `OnSuggestionDisplayed` should be fired for both predictors.
+                CheckThat(() => AssertDisplayedSuggestions(count: 2, predictorId_1, 56, 2)),
```
Comment (SteveL-MSFT) [inline test/ListPredictionTest.cs]: 56 here should also be MiniSessionid?

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: test/ListPredictionTest.cs (hunk 4 above)
Comment (daxian-dbw) [reply test/ListPredictionTest.cs]: Of course. My bad that I missed them :)

Comment (daxian-dbw) [reply test/ListPredictionTest.cs]: I just pushed the changes. Could you please review again when you have time, thanks!

**Review by SteveL-MSFT — APPROVED** [review]

**Comment by daxian-dbw** [comment]
The Aladdin team confirmed that the PSReadLine, PowerShell 7.2.0-preview.3, and Az.Predictor are working as expected end-to-end. I will merge this PR and get ready for the 2.2.0-beta2 release.

**Pull request merged**